Quadriga/src/main/java/edu/asu/spring/quadriga/rest/NetworkRestController.java

```diff
@@ -48,661 +50,636 @@
-				int latestVersion = networkManager.getLatestVersionOfNetwork(networkId) +1;
-				networkId=networkManager.storeNetworkDetails(res, user, networkName, network.getNetworkWorkspace().getWorkspace().getWorkspaceId(), INetworkManager.UPDATENETWORK, networkId,latestVersion);
-
-				if(networkId.endsWith(INetworkManager.DSPACEERROR)){
-					response.setStatus(404);
-					return "Text files don't belong to this workspace.";
-				}
-
-				response.setStatus(200);
-				response.setContentType(accept);
-				response.setHeader("networkid",networkId );
-				return res;
-			}
-		}
-	}
-
-
+    private static final Logger logger = LoggerFactory.getLogger(NetworkRestController.class);
+
+    @Autowired
+    private INetworkManager networkManager;
+
+    @Autowired
+    private IRestMessage errorMessageRest;
+
+    @Autowired
+    private IListWSManager wsManager;
+
+    @Autowired
+    private IEditorManager editorManager;
+
+    @Autowired
+    private IEditingNetworkAnnotationManager editingNetworkAnnoManager;
+
+    @Autowired
+    private IRestVelocityFactory restVelocityFactory;
+
+    @Autowired
+    private IUserManager userManager;
+
+    /**
+     * Rest interface for uploading XML for networks http://<<URL>:
+     * <PORT>>/quadriga/rest/uploadnetworks
+     * http://localhost:8080/quadriga/rest/uploadnetworks
+     * 
+     * @author Lohith Dwaraka
+     * @param request
+     * @param response
+     * @param xml
+     * @param accept
+     * @return XML
+     * @throws QuadrigaException
+     * @throws IOException
+     * @throws SAXException
+     * @throws ParserConfigurationException
+     * @throws JAXBException
+     * @throws TransformerException
+     * @throws QuadrigaStorageException
+     * @throws RestException
+     */
+    @RequestMapping(value = "rest/uploadnetworks", method = RequestMethod.POST)
+    public ResponseEntity<String> getNetworkFromClients(HttpServletRequest request, HttpServletResponse response,
+            @RequestBody String xml, @RequestHeader("Accept") String accept, Principal principal)
+                    throws QuadrigaException, ParserConfigurationException, SAXException, IOException, JAXBException,
+                    TransformerException, QuadrigaStorageException, RestException {
+
+        IUser user = userManager.getUser(principal.getName());
+        String networkName = request.getParameter("networkname");
+        String networkId = "";
+        String workspaceid = request.getParameter("workspaceid");
+
+        if (workspaceid == null || workspaceid.isEmpty()) {
+            String errorMsg = errorMessageRest.getErrorMsg("Please provide correct workspace id.");
+            return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        }
+        String projectid = networkManager.getProjectIdForWorkspaceId(workspaceid);
+        if (projectid == null || projectid.isEmpty()) {
+            String errorMsg = errorMessageRest.getErrorMsg("Please provide correct workspace id.");

```

Comment:
This is not the correct error message here. Should tell the user that the project id couldn't be found for the given workspace id (this should never happen).